### DISCUSSION TITLE
How to setup a MQTT broker

### DISCUSSION THREAD

**Discussion by crohot** [syssi/homeassistant-goecharger-mqtt#21] [Q&A]
Hi and thank you for this new approach.
Unfortunately, i have some issues: When setting up MQTT for the first time, i have to define a "Broker". Ist this some URL? Where would i find that? And do i have to define the same in the go-echarger app as well? Totally new to MQTT and the dependencies...
Cro.

**Comment by syssi** [comment]
The standard MQTT broker implementation is a linux daemon called `mosquitto` (`apt-get install mosquitto`). Per default the broker is listening on port 1883. The URL at the go-e app looks like this: `mqtt://hostname:1883` f.e. `mqtt://192.168.130.1:1883`.

If you add the MQTT integration manually at Home Assistant you have to connect to the same broker:

```
Broker: 192.168.130.1
Port: 1883
User: <optional>
Passwor: <optional>
```

If you use the Home Assistant Operation System there should be an `Add-on` called `Mosquitto broker` which can be instelled by one click:

https://github.com/home-assistant/addons/blob/master/mosquitto/README.md
https://smart-home-assistant.de/mqtt-server-installieren/
